Documentation/build-dependencies.rst

```diff
@@ -0,0 +1,321 @@
+..
+  SPDX-FileCopyrightText: 2018-2024 SeisSol Group
+
+  SPDX-License-Identifier: BSD-3-Clause
+
+.. _build_dependencies:
+
+Installing Dependencies
+=======================
+
+For compiling SeisSol, you will need the following dependencies during build:
+
+- A C++17-capable compiler. The following work:
+
+  - GCC (>= 9.0; tested: 13.2)
+  - ICX (tested: 2024.2)
+  - Clang (tested: 18, 19)
+  - NVHPC (tested: 24.09; currently still slow!)
+  - Cray CE (however: no commthread support; needs ``SEISSOL_COMMTHREAD=0``)
+  - ICC 2021.9 is supported up to v1.2.0 only
+- CMake (>= 3.20)
+- Python (>= 3.9)
+
+  - numpy (>= 1.12.0)
+  - setuptools (>= 0.61.0)
+
+(the CI currently verifies the build against Gcc 13.2, Clang 19, ICX 2024.2, and NVHPC 24.09)
+
+Additionally, you need the following libraries:
+
+- MPI (Support for MPI Standard >= 2.2)
+- Eigen (>= 3.4)
+- Hdf5 (>= 1.8, parallel)
+- easi (>= 1.5)
+- (optional, recommended) a code generator
+
+  - libxsmm (== 1.17 if using inline-assembly (LIBXSMM); otherwise >= 1.17 (LIBXSMM_JIT))
+  - PSpaMM
+- (optional, recommended) a mesh partitioning library
+
+  - ParMETIS (needs METIS and sometimes GKLib to be installed as well)
+  - ParHIP
+  - PT-SCOTCH
+- (optional, recommended) Netcdf (>= 4.4)
+- (optional, recommended) ASAGI
+
+All dependencies can be installed automatically with spack or manually.
+For the maximal performance and functionality,
+we always recommend installing all optional dependencies.
+(they can be skipped when pursuing a purely minimal installation)
+
+For the GPU version, the following packages need to be installed as well:
+
+- SYCL: either AdaptiveCpp (formerly known as hipSYCL/Open SYCL) >= 23.10 or DPC++
+- gemmforge (>= 0.0.207, for Nvidia, AMD and Intel GPUs)
+- (optional, recommended) chainforge (>= 0.0.3, for Nvidia and AMD GPUs)
+- CUDA (>= 11.0) for Nvidia GPUs, or HIP (ROCm >= 6.0.0) for AMD GPUs
+
+.. _spack_installation:
+
+Installation with Spack
+-----------------------
+
+The `Spack <https://github.com/spack/spack/wiki>`_ package ``seissol-env`` allows to automatically install all dependencies of SeisSol (e.g. mpi, hdf5, netcdf, easi, asagi, etc),
```

Comment:
shouldn't we better describe the seissol package?
(seissol-env package is only maintained because of the outdated spack on supermuc. they also have a new spack stack, but their tuning of the modules makes it difficult using it only for loading the dependencies, for latter manually compilation).

Reply:
some good but outdated usage of the spack package are heisenbug or frontera documentations.

Reply:
Ok, done.